visualisation/visualise-raw-data/visualise-raw-data-functions.R

```diff
@@ -0,0 +1,72 @@
+plot_raw_data = function(national = TRUE,
+                         cutoff = 25){
+  
+  # Get observed data ------------------------------------------------------------------
+  source(here::here("utils", "get-us-data.R"))
+  
+  daily_deaths_state <- get_us_deaths(data = "daily") %>%
+    dplyr::mutate(day = ordered(weekdays(as.Date(date)), 
+                                levels=c("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday")),
+                  epiweek_day = as.numeric(paste0(epiweek, ".", as.numeric(day)))) %>%
+    # filter out data from the last incomplete week
+    filter(epiweek_day < max(epiweek))
+  
+  
+  weekly_deaths_state <- daily_deaths_state %>%
+    dplyr::mutate(epiweek = lubridate::epiweek(date)) %>%
+    dplyr::group_by(state, epiweek) %>%
+    dplyr::summarise(deaths = sum(deaths),
+                     target_end_date = max(date)) %>%
+    dplyr::ungroup()
+  
+  weekly_deaths_national <- weekly_deaths_state %>%
+    dplyr::group_by(epiweek, target_end_date) %>%
+    dplyr::summarise(deaths = sum(deaths)) %>%
+    dplyr::ungroup()
+  
+  
+  
+  # Set observed data to match format
+  observed_deaths_state <- dplyr::filter(weekly_deaths_state) %>%
+    dplyr::mutate(model = "Observed") %>%
+    dplyr::select(-epiweek, c0.5 = deaths)
+  
+  observed_deaths_national <- weekly_deaths_national %>%
+    dplyr::mutate(model = "Observed",
+                  state = "US") %>%
+    dplyr::select(-epiweek, c0.5 = deaths)
+  
+  # Identify and filter which states to keep -------------------------------------------
+  
+  # Identify over 100 cases in the last week
+  source(here::here("utils", "states-min-last-week.R"))
+  keep_states <- states_min_last_week(min_last_week = cutoff, last_week = 1)
+  
+  
+  plot_state <- observed_deaths_state %>%
+    dplyr::filter(state %in% keep_states$state) 
+  
+  plot_national <- observed_deaths_national %>%
+    mutate(state = "US")
+  
+  plot_df <- 
```

Comment:
please no - assigning with an if else like this looks gross

Reply:
switch?

Reply:
anything else. 

if (something) {
assign
if else (something else) {
is fine
}